jpegenc/subblocks/block_buffer/block_buffer.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env python
+# coding=utf-8
+
+from myhdl import Signal, always_comb, always_seq, block, intbv, ResetSignal
+from jpegenc.subblocks.common import ram_in, ram_out, block_buffer_in, block_buffer_out
+from myhdl.conversion import analyze
+from math import log2, ceil
+from jpegenc.subblocks.block_buffer import RAM
+from jpegenc.subblocks.common import assign
+
+@block
+def block_buffer(inputs, outputs, clock, reset, line_width=16):
+
```

Comment:
The main idea is to use 3 block_buffers so, the video source will never stop the input of data. Each block buffer will consist of 8 rams and each ram will have the size of the line_width of the row.